cmd/internal/cli/key_pull.go

```diff
@@ -28,6 +29,22 @@ var KeyPullCmd = &cobra.Command{
 	DisableFlagsInUseLine: true,
 	PreRun:                sylabsToken,
 	Run: func(cmd *cobra.Command, args []string) {
+		// If flag did not change, look for a remote configuration, otherwise fall back to default
```

Comment:
And here. It might be possible to move this into a shared function, possibly taking a function argument to handle the different needs in each area.

Maybe something like:

```
func handleFlag(name string, fn func(bool)) {
    ...
}
```

That function could take the bool given by `cmd.Flags().Lookup("url").Changed`, or could take no arguments and just be run inside of that block. This function definitely needs to be fleshed out, but the idea is there maybe. Just something to keep in mind to maybe reduce duplication of efforts